lib/std/math/test/log1p.zig

```diff
@@ -0,0 +1,103 @@
+const math = @import("../../math.zig");
+const test_utils = @import("../test.zig");
+const Testcase = test_utils.Testcase;
+const runTests = test_utils.runTests;
+const floatFromBits = test_utils.floatFromBits;
+const negInf = test_utils.negInf;
+const nan32 = math.nan_f32;
+const nan64 = math.nan_f64;
+
+const Tc32 = Testcase(math.log1p, "log1p", f32);
+const tc32 = Tc32.init;
+
+const Tc64 = Testcase(math.log1p, "log1p", f64);
+const tc64 = Tc64.init;
+
+// Special-case tests shared between different float sizes, see genTests().
+const special_tests = .{
```

Comment:
For review its easier to write the special mathematical properties of log1p as set notation where log1p is defined as what value either above the test cases or near the top:
```txt
// log1p(a) - natural logarithm of one plus the input array, element-wise
in -> out
[-inf .. -min] -> -inf
..
[max .. inf] -> inf
```
Also order numbers from min to max input for consistency to minimize risk of leaving any gaps in the wanted test set.

Reply:
Ok, I'll add a comment at the top of each test file explaining the domain->range mapping. Sorting the tests seems sensible too.